app/controllers/UserProfileController.java

```diff
@@ -1,25 +1,39 @@
 package controllers;
 
+import org.sagebionetworks.bridge.BridgeConstants;
+import org.sagebionetworks.bridge.cache.ViewCache;
 import org.sagebionetworks.bridge.models.User;
 import org.sagebionetworks.bridge.models.UserProfile;
 import org.sagebionetworks.bridge.models.UserSession;
 import org.sagebionetworks.bridge.services.UserProfileService;
 
+import com.google.common.base.Supplier;
+
 import play.mvc.Result;
 
 public class UserProfileController extends BaseController {
     
     private UserProfileService userProfileService;
-
+    
+    private ViewCache viewCache;
+    
     public void setUserProfileService(UserProfileService userProfileService) {
         this.userProfileService = userProfileService;
     }
+    
+    public void setViewCache(ViewCache viewCache) {
+        this.viewCache = viewCache;
+    }
 
     public Result getUserProfile() throws Exception {
-        UserSession session = getAuthenticatedSession();
-
-        UserProfile profile = userProfileService.getProfile(session.getUser().getEmail());
-        return okResult(profile);
+        final UserSession session = getAuthenticatedSession();
+
+        String json = viewCache.getView(UserProfile.class, session.getUser().getEmail(), new Supplier<UserProfile>() {
```

Comment:
If we use email as the cache key, could this impact our ability to re-use email addresses across multiple studies?